syntax = "proto3";

option go_package=".;proto";

import "google/protobuf/empty.proto";

service Address {
    rpc QueryAddressList(AddressRequest)returns(AddressListResponse);
    rpc AddAddress(AddressRequest)returns(google.protobuf.Empty);
    rpc UpdateAddress(AddressRequest)returns(google.protobuf.Empty);
    rpc DeleteAddress(AddressRequest)returns(google.protobuf.Empty);
}

message AddressRequest{
    int32 id = 1;
    int32 user_id = 2;
    string province = 3;
    string city = 4;
    string district = 5;
    string address = 6;
    string signer_name = 7;
    string signer_mobile = 8;
    bool is_default = 9;
}

message AddressStatusResponse {
    int32 id = 1;
    bool is_default = 2;
}

message AddressDetailResponse{
    int32 id = 1;
    string province = 2;
    string city = 3;
    string district = 4;
    string address = 5;
    string signer_name = 6;
    string signer_mobile = 7;
    bool is_default = 8;
}

message AddressListResponse{
    int32 total = 1;
    repeated AddressDetailResponse data = 2;
}